src/xma/src/xmaapi/xmaadmin.cpp

```diff
@@ -177,6 +167,38 @@ xma_admin_session_create(XmaAdminProperties *props)
 
     session->base.session_signature = (void*)(((uint64_t)priv1) | ((uint64_t)priv1->reserved));
 
+    int32_t num_execbo = 6;
+    priv1->kernel_execbos.reserve(num_execbo);
+    priv1->num_execbo_allocated = num_execbo;
+    for (int32_t d = 0; d < num_execbo; d++) {
+        uint32_t  bo_handle;
+        int       execBO_size = MAX_EXECBO_BUFF_SIZE;
+        //uint32_t  execBO_flags = (1<<31);
+        char     *bo_data;
+        bo_handle = xclAllocBO(dev_handle, 
+                                execBO_size, 
+                                0, 
+                                XCL_BO_FLAGS_EXECBUF);
+        if (!bo_handle || bo_handle == mNullBO) 
+        {
+            xma_logmsg(XMA_ERROR_LOG, XMA_ADMIN_MOD,
+                    "Initalization of plugin failed. Failed to alloc execbo\n");
+            //Release singleton lock
+            g_xma_singleton->locked = false;
+            free(session->base.plugin_data);
```

Comment:
looks like this should be managed (RAII)?

Reply:
this is private space for plugins. xma only calloc